sql/core/src/test/scala/org/apache/spark/sql/TPCDSQuerySuite.scala

```diff
@@ -65,3 +67,8 @@ class TPCDSQuerySuite extends BenchmarkQueryTest with TPCDSBase {
 class TPCDSQueryWithStatsSuite extends TPCDSQuerySuite {
   override def injectStats: Boolean = true
 }
+
+class TPCDSQueryANSISuite extends TPCDSQuerySuite {
+  override protected def sparkConf: SparkConf =
+    super.sparkConf.set(SQLConf.ANSI_ENABLED, true)
+}
```

Comment:
Is this the only reason to remove these keywords? Currently users can still run TPCDS by turning off ANSI mode, right?

Reply:
It's the motivation, but the reason is these keywords are useless.

> users can still run TPCDS by turning off ANSI mode

Yes, but they can't with ANSI mode, isn't it a problem of ANSI mode?

Reply:
It sounds not a problem of ANSI mode or these queries. I think these queries just conflict with ANSI mode. To unblock TPCDS under ANSI mode, the change is okay.

The concern I have is that if we will remove more keywords from ANSI mode like this, then is ANSI mode still be necessary? This might be a rare case and the concern maybe not valid.

Reply:
I interpret ANSI mode as a better SQL dialect of Spark, which is ANSI compliant but not 100% the same as the ANSI standard. No database is 100% the same as ANSI, there are always extensions, e.g. unreserve some reserved keywords for better usability.

If a vector builds a product with spark on ANSI mode, and can't run TPCDS queries, I think the ANSI mode is problematic.

Reply:
Ok, I think this has a good reason to remove them. LGTM.